kato/kato-core/src/main/groovy/com/netflix/spinnaker/kato/helpers/OperationPoller.groovy

```diff
@@ -0,0 +1,93 @@
+package com.netflix.spinnaker.kato.helpers
+
+import com.netflix.spinnaker.kato.data.task.Task
+import com.netflix.spinnaker.kato.exception.OperationTimedOutException
+
+/**
+ * A poller with an upper time limit combined with a Fibonacci-based backoff.
+ * Let's you wrap the operation in a Groovy closure and the "if complete" operation in another.
+ * The "if complete" closure is fed the results of the first.
+ */
+class OperationPoller {
+
+  int asyncOperationTimeoutSecondsDefault
```

Comment:
I'd suggest making these final and giving this class a constructor, or slapping `@Immutable` on it